consensus/consensus-types/src/block.rs

```diff
@@ -300,6 +302,47 @@ impl Block {
             )
             .collect()
     }
+
+    fn new_block_metadata(&self, validators: &Vec<AccountAddress>) -> BlockMetadata {
+        let participants: HashSet<_> = self
+            .quorum_cert()
+            .ledger_info()
+            .signatures()
+            .keys()
+            .collect();
+
+        BlockMetadata::new(
+            self.id(),
+            self.epoch(),
+            self.round(),
+            self.voters_to_bitmap(validators, participants),
+            self.proposer_to_index(validators, self.author()),
+            self.timestamp_usecs(),
+        )
+    }
+
+    // TODO: as util function that can be tested with the reverse operation.
+    // TODO: also see functions in leader_reputation
+    // A bitmap of previous blocks, starting at 1
+    fn voters_to_bitmap(&self, validators: &Vec<AccountAddress>, voters: HashSet<&AccountAddress>) -> Vec<bool> {
```

Comment:
we probably want to group the conversion functions in a single place and have some unit tests for it, it seems very independent

Reply:
Do we have any examples in the codebase of what is the "rusty" way to do utility functions?